dotnet/src/SemanticKernel.Core/Orchestration/FunctionExecutor.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.SemanticKernel.Orchestration;
+
+/// <summary>
+/// Kernel function executor implementation
+/// </summary>
+internal sealed class FunctionExecutor : IFunctionExecutor
+{
+    private readonly Kernel _originKernel;
+
+    internal FunctionExecutor(Kernel originKernel)
+    {
+        this._originKernel = originKernel;
+    }
+
+    /// <inheritdoc/>
+    public Task<KernelResult> ExecuteAsync(ISKFunction skFunction, ContextVariables variables, IReadOnlyFunctionCollection? functions = null, CancellationToken cancellationToken = default)
+    {
+        using var kernel = this._originKernel.Clone(functions);
```

Comment:
Is this achieving anything?

E.g., when I load the semantic function it currently is bound to the kernel instance used to load it i.e. it has a reference to the kernel functions and gets the AI service from the kernel.

Reply:
The original idea was to expose a kernel that could be changed only during the execution time and those changes wouldn't affect the original kernel.